### PR TITLE
chore(Makefile): update go-dev to 0.10.0 and compress router binary

### PR DESCRIPTION
**Pull request by mboersma**
```
$ docker images
REPOSITORY                     TAG       IMAGE ID       CREATED             SIZE
192.168.64.2:5000/deis/router  after     c445c3459a68   4 minutes ago       38.14 MB
192.168.64.2:5000/deis/router  before    6b62b3583f0e   11 minutes ago      64.37 MB
`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 manifests/deis-router-rc.tmp.yaml
+rootfs/bin/router
```
Comment (krancour) [inline .gitignore]: I don't think this addition is required.  What is most likely is that you have a vestigial binary remaining in this old location from older build.  We used to put it there... but then it moved to its present location.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
...
 rootfs/opt/router/sbin/router
+rootfs/opt/router/sbin/router.*
```
Comment (krancour) [inline .gitignore]: Not sure I understand this addition.

File: .gitignore (hunk 1 above)
Comment (mboersma) [inline .gitignore]: You're right. I should have run `git clean`. :-)

File: .gitignore (hunk 2 above)
Comment (mboersma) [inline .gitignore]: `upx` created some temp binaries apparently. I had `router.upx` and `router.000` binaries there somehow while testing this PR.

Comment (krancour) [inline .gitignore]: Ok.  That makes sense.

**Comment by mboersma** [comment]
Updated to use [`deis/go-dev:0.9.1`](https://github.com/deis/docker-go-dev/releases/tag/0.9.1).

File: Makefile [hunk 3]
```diff
@@ -50,6 +53,8 @@ bootstrap: check-docker
 build: check-docker
 	mkdir -p ${BINDIR}
 	${DEV_ENV_CMD} make binary-build
+	$(call check-static-binary,$(BINDIR)/${SHORT_NAME})
+	${DEV_ENV_CMD} ${GOUPX} ${BINDIR}/${SHORT_NAME} || exit 1
```
Comment (sgoings) [inline Makefile]: Why do we need this `|| exit 1`?

**Comment by mboersma** [comment]
Updated to use [`deis/go-dev:0.10.0`](https://github.com/deis/docker-go-dev/releases/tag/0.10.0).

**Pull request merged**